# Humans_and_Lemurs_2017
This Repo contains scripts necessary to analyse and reproduced data for the Human and Lemur Metagenome ARG study

# Explanation of tables 
Tables are published under Version 2.0 on Zenodo [![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.10402843.svg)](https://doi.org/10.5281/zenodo.10402843)

File Name            |     Explanation        
------------------|--------------
Supplemental_1.txt | Deidentidfied epidemiological metadata for human samples
Supplemental_2.txt | MetaPhlAn 4 relative abundance estimates for all detectable bacteraia taxa
Supplemental_3.csv | MetaPhlAn 4 relative abundance estimates for all detectable taxa and unclassified reads 
Supplemental_4_v2.csv | contains read counts for detected antibiotic resistance genes with source and class metadata
Supplemental_5.csv | Output of AMRFinder for the assembled metagenomes
Supplemental_6.tab.txt | Bakta annotations of the assembled metagenomes
Supplemental_7.csv | All vs All BLAST output of assembled antibiotic resistance genes from humans and lemurs
Supplemental_8.csv | FastQC Output after host filtering
Supplemental_9.csv | Additional table merging lemur and human metadata
Supplemental_accessions.csv | List of SRA accessions for the analyzed metagenomes
refgenes.csv | The list of reference genes from NCBI's pathogen , version xxx. Contains AMR and virulence genes.
